ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -337,7 +347,7 @@ mod test {
         assert_eq!(uv_2, zip_chunks(U_2, V_2));
 
         // next iteration
-        let proof_2 = TestProofGenerator::compute_proof(uv_2.iter(), &lagrange_table);
+        let (proof_2, _) = TestProofGenerator::compute_proof(uv_2.iter(), &lagrange_table);
```

Comment:
To me, this line makes it so clear that this is the wrong design. We call `compute_proof` multiple times, but we only care about the size of the iterator we pass to it on the first iteration. Surely we should just treat that first iteration differently (e.g. add this counter to either the storage of multiplications to be validated, or possibly to the one-time conversion from binary to large field values).